src/devices/cpu/e132xs/e132xsop.hxx

```diff
@@ -1311,191 +1250,248 @@ void hyperstone_device::hyperstone_add_global_global()
 	m_icount -= m_clock_cycles_1;
 }
 
-void hyperstone_device::hyperstone_add_global_local()
+template <hyperstone_device::reg_bank DST_GLOBAL, hyperstone_device::reg_bank SRC_GLOBAL>
+void hyperstone_device::hyperstone_adds()
 {
-	check_delay_PC();
-
-	const uint32_t dst_code = DST_CODE;
-
-	const uint32_t sreg = m_local_regs[(SRC_CODE + GET_FP) & 0x3f];
-	uint32_t dreg = m_global_regs[dst_code];
-
-	const uint64_t tmp = (uint64_t)sreg + (uint64_t)dreg;
+	const uint32_t fp = GET_FP;
+	const uint32_t src_code = SRC_GLOBAL ? SRC_CODE : ((SRC_CODE + fp) & 0x3f);
+	const uint32_t dst_code = DST_GLOBAL ? DST_CODE : ((DST_CODE + fp) & 0x3f);
 
-	SR &= ~(C_MASK | V_MASK | Z_MASK | N_MASK);
+	const int32_t sreg = int32_t(SRC_GLOBAL ? ((src_code == SR_REGISTER) ? GET_C : m_global_regs[src_code]) : m_local_regs[src_code]);
+	int32_t dreg = int32_t((DST_GLOBAL ? m_global_regs : m_local_regs)[dst_code]);
+	const int64_t tmp = int64_t(sreg) + int64_t(dreg);
 
-	SR |= (tmp & 0x100000000L) >> 32;
+	SR &= ~(V_MASK | Z_MASK | N_MASK);
 	SR |= ((sreg ^ tmp) & (dreg ^ tmp) & 0x80000000) >> 28;
 
-	dreg += sreg;
-	set_global_register(dst_code, dreg);
+//#if SETCARRYS
+//  CHECK_C(tmp);
+//#endif
 
-	if (dst_code == 0)
-		SR &= ~M_MASK;
+	const int32_t res = sreg + dreg;
+	(DST_GLOBAL ? m_global_regs : m_local_regs)[dst_code] = res;
```

Comment:
This opcode handler assigns into `m_global_regs` directly rather than using set_global_register - is that intentional or an oversight?

Reply:
Oversight, can you fix?